### PR TITLE
Add speaker notes

### LINKED ISSUE #4: Speaker's notes
Find a system to display the speaker's notes in a way it's not annoying for viewers. Ideas:
- Use a popup and update the content once the section changes (`window.opener`).
- Use a button on each section to toggle speaker's notes on a side panel (but it displays them to anyone watching the keynote).

### PR DESCRIPTION
**Pull request by hissalht**
- Add a `notes` property to items of the `slides` collection.
- Render the notes in the notes page
- Add "goto" control buttons to change current slide
- Highlight current slide notes

TODO:
- Cleanup code
- Add some styling
- Use "cool" placeholder content

Closes #4

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -1,43 +1,75 @@
...
+    return acc;
+  }, {});
+
+  slides.forEach((slide) => {
+    slide.notes = notesBySlug[slide.fileSlug];
+  });
 }
+
+module.exports = function (config) {
+  config.addNunjucksFilter("noteSlug", (value) => {
```
Comment (bellangerq) [inline .eleventy.js:23]: I'm maybe missing something but can't we directly call `slide.fileSlug` in the template based on the fact that notes and slides have the same slug?

File: src/_includes/partials/intro.njk [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<header class="slide slide-intro">
+<header data-slide-slug="_intro_" class="slide slide-intro">
```
[the remark this answers is not in the document]
Comment (hissalht) [reply src/_includes/partials/intro.njk:1]: Oops, that's a remain from when I tried to use a "notes" collection and I needed to remove the ".notes" part of the slug.

It is not needed anymore indeed.

File: .eleventy.js (hunk 1 above)
Comment (hissalht) [reply .eleventy.js:23]: Oops, that's a remain from when I tried to use a "notes" collection and I needed to remove the ".notes" part of the slug.

It is not needed anymore indeed.

File: .eleventy.js [hunk 3]
```diff
@@ -1,43 +1,75 @@
...
-			output: 'dist'
-		}
-	}
+/**
+ * Add a notes property to the slide templates objects.
+ * @param {object[]} slides
+ * @param {object[]} notes
+ */
+function linkSlidesWithNotes(slides, notes) {
+  const notesBySlug = notes.reduce((acc, note) => {
```
[the remark this answers is not in the document]
Comment (hissalht) [reply .eleventy.js:11]: 👌

File: src/scripts/notes.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+let currentSlideIndex;
+
+function updateSlideIndex(slug) {
+  currentSlideIndex = slug;
+  const slideContent = document.getElementById("current-slide");
+  slideContent.innerHTML = slug;
+
+  document.querySelectorAll("section[data-note-slug]").forEach((el) => {
```
[the remark this answers is not in the document]
Comment (hissalht) [reply src/scripts/notes.js:8]: 👌

File: src/_includes/partials/intro.njk (hunk 2 above)
[the remark this answers is not in the document]
Comment (hissalht) [reply src/_includes/partials/intro.njk:1]: I think `_intro_` makes it obvious that it's a "special" slide. I'm not a fan of restricting a specific keyword for naming files, the user will just go around it and name their slide `introduction.md` or something

[the remark this answers is not in the document]
Comment (bellangerq) [reply src/_includes/partials/intro.njk:1]: Ok!

**Pull request merged**